examples/permissioned-token/test/common.js

```diff
@@ -0,0 +1,380 @@
+    toAccountAddr: opAccount.address,
+    assetID: assetIndex,
+    amount: 0,
+    payFlags: { totalFee: 1000, closeRemainderTo: manager.address }
+  };
+  runtime.executeTx(optOutParams);
+}
+
+/**
+ * Force Transfer Tokens
+ * @param runtime RuntimeEnv Instance
+ * @param from Sender Account
+ * @param to Receiver Account
+ * @param amount Amount
+ * @param assetIndex ASA ID
+ * @param controllerAppID Controller App ID
+ * @param permissionsAppId Permissions App ID
+ * @param lsig Clawback LSig
+ * @param manager ASA Manager Account
+ */
+function forceTransfer (
+  runtime, from, to, amount, assetIndex, controllerAppID, permissionsAppId, lsig, manager) {
+  const txGroup = [
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: manager.account,
+      appId: controllerAppID,
+      payFlags: { totalFee: 1000 },
+      appArgs: ['str:force_transfer'],
+      accounts: [to.address],
+      foreignAssets: [assetIndex]
+    },
+    {
+      type: types.TransactionType.RevokeAsset,
+      sign: types.SignType.LogicSignature,
+      fromAccountAddr: lsig.address(),
+      recipient: to.address,
+      assetID: assetIndex,
+      revocationTarget: from.address,
+      amount: amount,
+      lsig: lsig,
+      payFlags: { totalFee: 1000 }
+    },
+    {
+      type: types.TransactionType.TransferAlgo,
+      sign: types.SignType.SecretKey,
+      fromAccount: manager.account,
+      toAccountAddr: lsig.address(),
+      amountMicroAlgos: 1000,
+      payFlags: { totalFee: 1000 }
+    },
+    {
+      type: types.TransactionType.CallNoOpSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: manager.account,
+      appId: permissionsAppId,
+      payFlags: { totalFee: 1000 },
+      appArgs: [TRANSFER],
+      accounts: [from.address, to.address]
+    }
+  ];
+  runtime.executeTx(txGroup);
+}
+
+function setupEnv () {
+  // Create Accounts and Env
+  alice = new AccountStore(minBalance, { addr: ALICE_ADDRESS, sk: new Uint8Array(0) });
+  bob = new AccountStore(minBalance);
+  elon = new AccountStore(minBalance);
+  runtime = new Runtime([master, alice, bob, elon]);
+
+  // Deploy ASA
+  assetIndex = runtime.addAsset('gold', { creator: { ...alice.account, name: 'alice' } });
+  const asaDef = runtime.getAssetDef(assetIndex);
+
+  // Deploy Controller SSC
+  let sscFlags = {
+    sender: alice.account,
+    localInts: 0,
+    localBytes: 0,
+    globalInts: 2,
+    globalBytes: 0,
+    appArgs: [`int:${assetIndex}`],
+    foreignAssets: [assetIndex]
+  };
+  CONTROLLER_PROGRAM = getProgram(CONTROLLER, { TOKEN_ID: assetIndex });
+  controllerAppID = runtime.addApp(
+    sscFlags, {}, CONTROLLER_PROGRAM, CLEAR_STATE_PROGRAM
+  );
+
+  // Deploy Clawback Lsig and Modify Asset
+  CLAWBACK_PROGRAM = getProgram(CLAWBACK, {
+    TOKEN_ID: assetIndex,
+    CONTROLLER_APP_ID: controllerAppID
+  });
+  lsig = runtime.getLogicSig(CLAWBACK_PROGRAM, []);
+  fund(runtime, master, lsig.address());
```

Comment:
Why do we "fund" an account here? Aren't we intializing these accounts with some `minBalance`?

Reply:
We are funding lsig here.